internal/util/ArgoUtil/ClusterService.go

```diff
@@ -21,28 +21,49 @@ import (
 	"github.com/argoproj/argo-cd/v2/pkg/apis/application/v1alpha1"
 )
 
-type ClusterService interface {
+type ClusterConfig struct {
+	BearerToken     string `json:"bearerToken"`
+	TLSClientConfig `json:"tlsClientConfig"`
+}
+type TLSClientConfig struct {
+	Insecure bool `json:"insecure"`
+}
+type ClusterRequest struct {
+	Name   string        `json:"name"`
+	Server string        `json:"server"`
+	Config ClusterConfig `json:"config"`
+}
+type ClusterResponse struct {
+	Id            string        `json:"id"`
+	Server        string        `json:"server"`
+	Name          string        `json:"name"`
+	ServerVersion string        `json:"serverVersion"`
+	Config        ClusterConfig `json:"config"`
+}
+
+type ArgoClusterService interface {
 	GetClusterByServer(server string) (*v1alpha1.Cluster, error)
 	ClusterList() (*v1alpha1.ClusterList, error)
 	CreateCluster(cluster v1alpha1.Cluster) (*v1alpha1.Cluster, error)
+	CreateClusterWithGitposConfigured(request *ClusterRequest, acdToken string) (*ClusterRequest, error)
```

Comment:
function name CreateClusterWithGitposConfigured doesn't make much sense, rename it

Reply:
done